ci-operator/step-registry/telcov10n/sno-hub-ztp/set-hub-preserved/telcov10n-sno-hub-ztp-set-hub-preserved-commands.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ telcov10n cluster setup via agent command ************"
+# Fix user IDs in a container
+[ -e "${HOME}/fix_uid.sh" ] && "${HOME}/fix_uid.sh" || echo "${HOME}/fix_uid.sh was not found" >&2
+
+function setup_aux_host_ssh_access {
+
+  echo "************ telcov10n Setup AUX_HOST SSH access ************"
+
+  SSHOPTS=(
+    -o 'ConnectTimeout=5'
+    -o 'StrictHostKeyChecking=no'
+    -o 'UserKnownHostsFile=/dev/null'
+    -o 'ServerAliveInterval=90'
+    -o LogLevel=ERROR
+    -i "${CLUSTER_PROFILE_DIR}/ssh-key"
+  )
+
+}
+
+function set_expiration_time {
+
+  echo "************ telcov10n Set Hub cluster expiration time ************"
+
+  echo
+  set -x
+  b64_exp_time=$(echo "${EXPIRATION_TIME}" | base64 -w 0)
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s --  \
+  "${NAMESPACE}" "${b64_exp_time}" "${EXPIRATION_TIME_FILE}" "${SHARED_HUB_CLUSTER_PROFILE}" << 'EOF'
+set -o nounset
+set -o errexit
+set -o pipefail
+
+exp_time_ns_file=/var/builds/${1}
+exp_time=$(echo ${2} | base64 -d)
+shared_exp_time_ns_file_sym_link=${3}
+cluster_profile_folder=$(dirname ${exp_time_ns_file})/${1}-cluster_profile_dir
+shared_cluster_profile_folder=${4}
+
+mkdir -pv $(dirname ${exp_time_ns_file})
+mkdir -pv ${cluster_profile_folder}
+
+TZ=UTC date --iso-8601=seconds -d "${exp_time}" > ${exp_time_ns_file}
+
+ln -sf ${exp_time_ns_file} ${shared_exp_time_ns_file_sym_link}
+ln -sf ${cluster_profile_folder} ${shared_cluster_profile_folder}
+
+echo "Now       : $(TZ=UTC date --iso-8601=seconds)"
```

Comment:
this too.. limit the execution of remote commands to the minimum... Eventually, we would have an API and avoid ssh at all.